### ISSUE TITLE
Remove support of Zimit1 ZIM files

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#3910]
We don't distribute such ZIM files anymore which have been almost all superseeded by Zimit2 ZIM files which don't rely anymore on ServiceWorker. Therefore all the tooling to get the ZIMit1 files is not necessary anymore and should be removed.

**Comment by kelson42** [comment]
@MohitMaliFtechiz @rgaudin The main question is when exactly? I see no urgency here but I would like to ask you if this feature/tooling has an impact for normal ZIM reading? Does it somehow slowdown the read of all "normal" ZIM?

**Comment by rgaudin** [comment]
From what I recall it was quite invasive.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 It does not slow the reading of normal zim files, the service worker only provides the power to intercept the request of the service worker and we implement our own logic to load the data. If no service worker is available in any zim, the normal webViewClient callback gives us the same thing to load our own data. There is no slowdown in reading the normal zim files(Since they already loading via the webViewClient callback).

**Comment by Jaifroid** [comment]
I'd urge a bit of caution and that you allow some time to ensure all the old Zimit1 archives have been successfully republished as Zimit2 (maybe that's already happened, but I'm not sure it's 100%). People will still have old Zimit1 archives that they cherish, and may want to keep them for some time. Do you really want to render them unreadable with the Android app, which was one of the only ways in which a Zimit1 archive could be successfully read? I know I've still got some that I don't think have been successfully replaced yet. E.g. the CIA World Factbook. It's no longer available in the zimit folder, but there is no alternative either... FYI I don't plan to deprecate support for zimit1 in KJS yet...

**Comment by MohitMaliFtechiz** [comment]
@Jaifroid I agree with your point. We should not remove the zimit1 support from the application as users can have these files. Maybe we can consider this ticket in the future when all the users are switched to zimit2, but for now, we should keep the support for zimit1.

**Comment by kelson42** [comment]
Almost two years have passed now. All the old zimit1 files have been refreshed. We are about to release a major release, this is the moment to remove this.

**Comment by Jaifroid** [comment]
CIA World Factbook is still only available in classic Zimit format, as just one example of many.... Just sayin' 🤷

**Comment by kelson42** [comment]
> CIA World Factbook is still only available in classic Zimit format, as just one example of many.... Just sayin' 🤷

Do we distribute this file? I don't see it in the library https://library.kiwix.org/#lang=eng&q=cia

Do you have other example?

**Comment by Jaifroid** [comment]
No we don't distribute it, but people may have it and there isn't an updated version. Depends whether you want the Android app to be able to read historical archives or not. I'm just being the voice of caution, but it's your decision of course! @benoit74 might know if there are any Zimit 1 archives actually still available for download without a newer version being available.

You can also decide to let the PWA and the Browser Extensions be the recommended ways to read such historical archives. I don't plan to remove that code, as it would be a lot of untangling for no extra benefit (code only runs when needed).

**Comment by benoit74** [comment]
We (Kiwix/openZIM) do not provide anymore Zimit1 ZIMs since years (afaik). But it is true that some titles have not been updated to Zimit 2 (e.g. https://farm.openzim.org/recipes/developer.mozilla.org_en for which we still do not have a fix - only a hack I do not even really remember about) and we might have others in the wild.

I don't know if it is a sufficient reason to not drop support for Zimit 1 ZIM files. 

For me the decision should be based on:
- Zimit1 support has significant perf impact? (answer so far is "no")
- Zimit1 support has caused bugs? or is it even known to be significantly buggy which would cause bad user experience?
- Zimit1 support has significant impact on code readability? on app maintenance cost? on Google validation?

If not, I would keep the feature as long as possible

**Comment by kelson42** [comment]
@MohitMaliFtechiz Please drop now for the reasons given: deprecated, broken, unused, invasive code.

**Comment by ferrouswheel** [comment]
CIA World Fact Book was just [deleted by the CIA](https://www.cia.gov/stories/story/spotlighting-the-world-factbook-as-we-bid-a-fond-farewell/) - so anyone that has the zimit1 version now has something important for archival purposes. Is there way to convert Zimit 1 to the new format for preservation and future proofing?

The World Fact Book was [announced on the kiwix blog in 2023](https://kiwix.org/en/kiwix-content-updates-1/)

**Comment by kelson42** [comment]
@ferrouswheel Sad news but AFAIK there is no way. See also https://github.com/openzim/zim-requests/issues/988
